code/drasil-gool/lib/Drasil/GOOL/LanguageRenderer/LanguagePolymorphic.hs

```diff
@@ -134,16 +134,26 @@ smartAdd v1 v2 = do
   v1' <- v1
   v2' <- v2
   case (RC.valueInt v1', RC.valueInt v2') of
-    (Just i1, Just i2) -> litInt (i1 + i2)
-    _                  -> v1 #+ v2
+    (Just i1,Just i2) -> litInt (i1 + i2)
```

Comment:
I wonder if `smartAdd` should be "more typed", so that this cascade of checking doesn't happen.